code/game/objects/structures/crates_lockers/closets/cardboardbox.dm

```diff
@@ -62,3 +64,34 @@
 			for(var/mob/M in viewers(src))
 				M.show_message("<span class='notice'>\The [src] has been cut apart by [user] with \the [WC].</span>", 3, "You hear cutting.", 2)
 			qdel(src)
+			return
+		if(istype(W, /obj/item/weapon/storage/fancy/crayons))
+			var/list/contents = W.contents
+			var/crayoncount = 0
+			for(var/C in contents)
+				if(istype(C, /obj/item/toy/crayon))
+					crayoncount += 1
+				continue
+			if(crayoncount == 6)
+				var/decalselection = input("Please select a decal") as null|anything in list("Atmospherics", "Bartender", "Barber", "Blueshield",	"Brig Physician", "Captain",
+					"Cargo", "Chief Engineer",	"Chaplain",	"Chef", "Chemist", "Civilian", "Clown", "CMO", "Coroner", "Detective", "Engineering", "Genetics", "HOP",
+					"HOS", "Hydroponics", "Internal Affairs Agent", "Janitor",	"Magistrate", "Mechanic", "Medical", "Mime", "Mining", "NT Representative", "Paramedic", "Pod Pilot",
+					"Prisoner",	"Research Director", "Security", "Syndicate", "Therapist", "Virology", "Warden", "Xenobiology")
+				if(!decalselection)
+					return
+				if(user.incapacitated())
+					return
+				if(W == user.get_active_hand() && Adjacent(usr))
```

Comment:
Your code doesn't end up getting *too* deep, but whenever possible, you want to bail out instead of adding another indentation level. Just like the checks immediately above this, you can format this with a return:
```
if(!(W == user.get_active_hand() && Adjacent(user)))
  // Side note: don't use "usr" when you should be using "user"
  return
```
This becomes particularly important when you want to do *more* than just return, because you'd end up having to add `else` statements at specific indentation levels way below the rest of the code, which is a readability nightmare. Instead, you'll have something like...
```
if(!enoughcrayons)
  to_chat(user, "oh no you require additional crayons")
  return
```
Some of your failure states - such as the first check for having enough crayons - should have a message like that to explain what actually went wrong. Obvious stuff (canceling, moving away from the box, etc) doesn't, of course.